tests/PHPStan/Analyser/data/array-map-callable81.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace ArrayMapCallable81;
+
+use function PHPStan\Testing\assertType;
+use function strval as str;
+
+class Foo
+{
+	/**
+	 * @template T of int
+	 * @param T $n
+	 * @return (T is 3 ? 'Fizz' : (T is 5 ? 'Buzz' : T))
+	 */
+	public static function fizzbuzz(int $n): int|string
+	{
+		return match ($n) {
+			3 => 'Fizz',
+			5 => 'Buzz',
+			default => $n,
+		};
+	}
+
+	public function doFoo(): void
+	{
+		$a = range(0, 1);
+
+		assertType("array{'0', '1'}", array_map('strval', $a));
+		assertType("array{'0', '1'}", array_map(strval(...), $a));
+		assertType("array{'0', '1'}", array_map(str(...), $a));
+		assertType("array{'0'|'1', '0'|'1'}", array_map(fn ($v) => strval($v), $a));
+		assertType("array{'0'|'1', '0'|'1'}", array_map(fn ($v) => (string)$v, $a));
```

Comment:
The `ClosureType::$returnType` is bound to the union of the other elements of the array with the type of the callback. I'm looking for a way to get around that.